BraintreeCore/src/main/java/com/braintreepayments/api/Configuration.kt

```diff
@@ -0,0 +1,410 @@
+        googlePayConfiguration = GooglePayConfiguration.fromJson(json.optJSONObject(GOOGLE_PAY_KEY))
+        isThreeDSecureEnabled = json.optBoolean(THREE_D_SECURE_ENABLED_KEY, false)
+        venmoConfiguration = VenmoConfiguration.fromJson(json.optJSONObject(PAY_WITH_VENMO_KEY))
+        kountConfiguration = KountConfiguration.fromJson(json.optJSONObject(KOUNT_KEY))
+        unionPayConfiguration = UnionPayConfiguration.fromJson(json.optJSONObject(UNIONPAY_KEY))
+        visaCheckoutConfiguration =
+            VisaCheckoutConfiguration.fromJson(json.optJSONObject(VISA_CHECKOUT_KEY))
+        graphQLConfiguration = GraphQLConfiguration.fromJson(json.optJSONObject(GRAPHQL_KEY))
+        samsungPayConfiguration =
+            SamsungPayConfiguration.fromJson(json.optJSONObject(SAMSUNG_PAY_KEY))
+        cardinalAuthenticationJwt = Json.optString(json, CARDINAL_AUTHENTICATION_JWT, null)
+    }
+
+    /**
+     * @return `true` if cvv is required for card transactions, `false` otherwise.
+     */
+    open val isCvvChallengePresent = challenges.contains("cvv")
+
+    /**
+     * @return `true` if postal code is required for card transactions, `false` otherwise.
+     */
+    open val isPostalCodeChallengePresent = challenges.contains("postal_code")
+
+    /**
+     * @return `true` if fraud device data collection should occur; `false` otherwise.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    val isFraudDataCollectionEnabled = cardConfiguration.isFraudDataCollectionEnabled
+
+    /**
+     * @return `true` if Venmo is enabled for the merchant account; `false` otherwise.
+     */
+    open val isVenmoEnabled = venmoConfiguration.isAccessTokenValid
+
+    /**
+     * @return the Access Token used by the Venmo app to tokenize on behalf of the merchant.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    val venmoAccessToken = venmoConfiguration.accessToken
+
+    /**
+     * @return the Venmo merchant id used by the Venmo app to authorize payment.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    val venmoMerchantId = venmoConfiguration.merchantId
+
+    /**
+     * @return the Venmo environment used to handle this payment.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    val venmoEnvironment = venmoConfiguration.environment
+
+    /**
+     * @return `true` if GraphQL is enabled for the merchant account; `false` otherwise.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    val isGraphQLEnabled = graphQLConfiguration.isEnabled
+
+    /**
+     * @return `true` if Local Payment is enabled for the merchant account; `false` otherwise.
+     */
+    open val isLocalPaymentEnabled = isPayPalEnabled // Local Payments are enabled when PayPal is enabled
+
+    /**
+     * @return `true` if Kount is enabled for the merchant account; `false` otherwise.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
```

Comment:
This is neat!

Reply:
What is `LIBRARY_GROUP`? How broad of a restriction is that?

Reply:
https://developer.android.com/reference/androidx/annotation/RestrictTo.Scope

I wonder if this is just a docs annotation to hide the value in the reference docs, or if it actually prevents the value from being accessed outside the "library group". I beleive the later.

Reply:
It's also restricts access to libraries with the group id `com.braintreepayments.com` so basically any of the libraries we develop have access e.g `com.braintreepayments.com:card` can talk to `com.braintreepayments.com:braintree-core`, etc.